api/wasm/cpp/Makefile.base

```diff
@@ -6,10 +6,9 @@ else
   CPP_CONTEXT_LIB = ${CPP_API}/proxy_wasm_intrinsics.cc
 endif
 
-%.wasm %.wat: %.cc ${CPP_API}/proxy_wasm_intrinsics.h ${CPP_API}/proxy_wasm_enums.h ${CPP_API}/proxy_wasm_externs.h ${CPP_API}/proxy_wasm_impl.h ${CPP_API}/proxy_wasm_intrinsics.js ${CPP_CONTEXT_LIB}
+%.wasm %.wat: %.cc ${CPP_API}/proxy_wasm_intrinsics.h ${CPP_API}/proxy_wasm_enums.h ${CPP_API}/proxy_wasm_externs.h ${CPP_API}/proxy_wasm_api.h ${CPP_API}/proxy_wasm_intrinsics.js ${CPP_CONTEXT_LIB}
 	em++ -s WASM=1 -s BINARYEN_TRAP_MODE='clamp' -s LEGALIZE_JS_FFI=0 -s EMIT_EMSCRIPTEN_METADATA=1 --std=c++17 -O3 -g3 -I${CPP_API} -I/usr/local/include --js-library ${CPP_API}/proxy_wasm_intrinsics.js $*.cc ${CPP_API}/proxy_wasm_intrinsics.pb.cc ${CPP_CONTEXT_LIB} ${CPP_API}/libprotobuf.bc -o $*.js
 	wavm-disas $*.wasm $*.wat
-	wavm-compile $*.wasm $*.wasm
```

Comment:
Could you split to this another PR? Ideally, also removing `allow_precompiled` from the codebase (assuming that's the intention here) and relevant tests for it? There is also one more use of `wavm-compile` in `test/extensions/wasm/test_data/Makefile`.

Reply:
https://github.com/envoyproxy/envoy-wasm/pull/135

Reply:
I think we do want to allow precompiling since that could be a big win for WAVM.   However it will come with some warnings about using the right version of WAVM.